client/signup/steps/import-url/index.jsx

```diff
@@ -15,34 +17,48 @@ import Button from 'components/button';
 import ButtonGroup from 'components/button-group';
 import StepWrapper from 'signup/step-wrapper';
 import SignupActions from 'lib/signup/actions';
-import FormTextInputWithAction from 'components/forms/form-text-input-with-action';
-//import FormInputValidation from 'components/forms/form-input-validation';
-import { fetchIsSiteImportable, setNuxUrlInputValue } from 'state/importer-nux/actions';
+import FormTextInput from 'components/forms/form-text-input';
+import FormInputValidation from 'components/forms/form-input-validation';
+import { setNuxUrlInputValue, setValidationMessage } from 'state/importer-nux/actions';
 import {
 	getNuxUrlInputValue,
 	getSiteDetails,
+	getUrlInputValidationMessage,
 	isUrlInputDisabled,
 } from 'state/importer-nux/temp-selectors';
 
 const debug = debugFactory( 'calypso:signup-step-import-url' );
 
-const DEBOUNCE_CHECK_INTERVAL = 1200;
+const VALIDATION_INTERVAL = 1200;
 
 const normalizeUrlForImportSource = url => {
 	// @TODO sanitize? Prepend https:// ..?
 	return url;
 };
 
+const isValidUrl = ( value = '' ) => {
+	const { protocol } = parseURL( value );
+	const withProtocol = protocol ? value : 'http://' + value;
+	const { hostname } = parseURL( withProtocol );
+
+	return isWebUri( withProtocol ) && indexOf( hostname, '.' ) > 0;
+};
+
 class ImportURLStepComponent extends Component {
 	componentDidMount() {
 		const { queryObject } = this.props;
 		const urlFromQueryArg = normalizeUrlForImportSource( get( queryObject, 'url' ) );
 
 		if ( urlFromQueryArg ) {
-			this.fetchIsSiteImportable( urlFromQueryArg );
+			this.props.setNuxUrlInputValue( urlFromQueryArg );
+			defer( this.checkValidation );
 		}
 	}
 
+	getValidationMessage( value ) {
```

Comment:
A thought I had here (but only after writing all of this) was perhaps to just set `isValid` in state as a boolean, and show the message based on that boolean.
That would neaten things up and be more semantic but would leave us unable to be more nuanced with our messaging in the future.
Any thoughts there?

Reply:
I wouldn't go for a boolean, as we already know there will be a few more wix-specific validations that I will port from the old import UI. 

I think a message is fine here, maybe error code if we want to make it more semantic and use switch to decide what message/component should get rendered for each code.